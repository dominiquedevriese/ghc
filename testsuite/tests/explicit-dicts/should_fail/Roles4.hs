{-# LANGUAGE RoleAnnotations #-}
{-# LANGUAGE TypeApplications #-}
module Roles4 where

import GHC.Base (Ord(..))

type role Set nominal
data Set a

reverseOrd :: Ord a => Ord.Dict a
reverseOrd = undefined

f4 = (undefined :: Ord a => f a -> Set (f a) -> Set (f a))
     ((reverseOrd @(Maybe Int) :: Ord a))
