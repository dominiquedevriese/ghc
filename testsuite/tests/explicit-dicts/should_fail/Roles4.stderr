
Roles4.hs:13:6: error:
    • Explicit dictionary application to: ‘Ord a’
      is not allowed because of instance incoherence:
      In ‘f a -> Set (f a) -> Set (f a)’
        Type variable ‘a’ has role Nominal
    • In the expression:
        (undefined :: Ord a => f a -> Set (f a) -> Set (f a))
          ((reverseOrd @(Maybe Int) :: Ord a))
      In an equation for ‘f4’:
          f4
            = (undefined :: Ord a => f a -> Set (f a) -> Set (f a))
                ((reverseOrd @(Maybe Int) :: Ord a))
